x-pack/plugins/entity_manager/server/lib/queries/index.test.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getEntityInstancesQuery } from '.';
+
+describe(__filename, () => {
+  describe('getEntityInstancesQuery', () => {
+    it('generates a valid esql query', () => {
+      const query = getEntityInstancesQuery({
+        source: {
+          type: 'service',
+          index_patterns: ['logs-*', 'metrics-*'],
+          identity_fields: ['service.name'],
+          metadata_fields: ['host.name'],
+          filters: [],
+          timestamp_field: 'custom_timestamp_field',
+        },
+        limit: 5,
+        start: '2024-11-20T19:00:00.000Z',
+        end: '2024-11-20T20:00:00.000Z',
+      });
+
+      expect(query).toEqual(
```

Comment:
I wonder if there is any tooling within Kibana to validate the syntax of this query automatically.

Reply:
I've logged this in https://github.com/elastic/elastic-entity-model/issues/203